cpp/include/Ice/ClientAuthenticationOptions.h

```diff
@@ -0,0 +1,414 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#ifndef ICE_SSL_CLIENT_AUTHENTICATION_OPTIONS_H
+#define ICE_SSL_CLIENT_AUTHENTICATION_OPTIONS_H
+
+#include "SSLConfig.h"
+#include "SSLConnectionInfo.h"
+
+#include <functional>
+
+namespace Ice::SSL
+{
+    /**
+     * The SSL configuration properties for client connections.
+     */
+    struct ClientAuthenticationOptions
+    {
+#if defined(_WIN32)
+        /**
+         * A callback that allows selecting the client's SSL certificate based on the target server host name.
+         *
+         * @remarks This callback is invoked by the SSL transport for each new outgoing connection before starting the
+         * SSL handshake to determine the appropriate client certificate. The callback should return a PCCERT_CONTEXT
+         * that represents the client's certificate. The SSL transport takes ownership of the returned certificate and
+         * releases it when the connection is closed.
+         *
+         * @param host The target server host name.
+         * @return The client's certificate, or nullptr to indicate that no certificate is used by the connection.
+         *
+         * Example of setting clientCertificateSelectionCallback:
+         * ```cpp
+         * PCCERT_CONTEXT _clientCertificate = ...; // Load the client certificate using WinCrypt API
+         *
+         * auto initData = Ice::InitializationData {
+         *   ...
+         *   .clientAuthenticationOptions = ClientAuthenticationOptions {
+         *      .clientCertificateSelectionCallback = [this](const std::string&)
```

Comment:
I find the code somewhat unclear. It's not obvious we capture "this" to use the _clientCertificate field.

Reply:
ok, I plan to move this example to separate C++ source files and use `@include`, it is a pain to write this in the comment directly.